src/runtime/virtcontainers/sandbox.go

```diff
@@ -615,8 +615,19 @@ func newSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Factor
 	sandboxConfig.HypervisorConfig.RunStorePath = s.store.RunStoragePath()
 
 	spec := s.GetPatchedOCISpec()
-	if spec != nil && spec.Process.SelinuxLabel != "" {
-		sandboxConfig.HypervisorConfig.SELinuxProcessLabel = spec.Process.SelinuxLabel
+	if spec != nil {
+		if sandboxConfig.HypervisorConfig.DisableGuestSeLinux {
```

Comment:
IMHO, this will break SELinux integration on the host as DisableGuestSelinux is set to `true` by default, meaning the `else if` would basically be a dead code.

More than that, we're mixing 2 things here:
* SELinux on the host, where we set `sandboxConfig.HypervisorConfig.SELinuxProcessLabel = spec.Process.SelinuxLabel`
* SELinux on the guest, where you check for DisableGuestSelinux.

And it seems the wrong place to do so.

Reply:
Am I understanding you correctly that you would not have me reset the `SELinuxProcessLabel` and only touch the spec itself?

Reply:
For me, I only care about the MountLabel, as this was what broke when applied by the `kata-agent` from inside the VM when creating the sandbox. I can remove resetting the `Process.SelinuxLabel` field if this would change things.

Reply:
@mikn there's code already to control SELinux labels that should be passed to the guest at https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/kata_agent.go#L994.

Can you share your configuration.toml file ?

Reply:
Ah - okay. This seems to only affect `runtime-rs`. I completely missed this code path when reviewing the Go version. I apologise for wasting your time! I will back my Go fixes and leave only the rust fixes.